src/components/chat-view/chat-input/LexicalContentEditable.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  InitialConfigType,
+  InitialEditorStateType,
+  LexicalComposer,
+} from '@lexical/react/LexicalComposer'
+import { ContentEditable } from '@lexical/react/LexicalContentEditable'
+import { EditorRefPlugin } from '@lexical/react/LexicalEditorRefPlugin'
+import { LexicalErrorBoundary } from '@lexical/react/LexicalErrorBoundary'
+import { HistoryPlugin } from '@lexical/react/LexicalHistoryPlugin'
+import { OnChangePlugin } from '@lexical/react/LexicalOnChangePlugin'
+import { RichTextPlugin } from '@lexical/react/LexicalRichTextPlugin'
+import { LexicalEditor, SerializedEditorState } from 'lexical'
+import { RefObject, useCallback } from 'react'
+
+import { useApp } from '../../../contexts/app-context'
+import { fuzzySearch } from '../../../utils/fuzzy-search'
+
+import OnVaultChatCommandPlugin from './on-vault-chat/OnVaultChatPlugin'
+import AutoFocusPlugin from './plugins/auto-focus/AutoFocusPlugin'
+import AutoLinkMentionPlugin from './plugins/mention/AutoLinkMentionPlugin'
+import { MentionNode } from './plugins/mention/MentionNode'
+import MentionPlugin from './plugins/mention/MentionPlugin'
+import NoFormatPlugin from './plugins/no-format/NoFormatPlugin'
+import OnEnterPlugin from './plugins/on-enter/OnEnterPlugin'
+import OnMutationPlugin, {
+  NodeMutations,
+} from './plugins/on-mutation/OnMutationPlugin'
+import CreateTemplatePopoverPlugin from './plugins/template/CreateTemplatePopoverPlugin'
+import TemplatePlugin from './plugins/template/TemplatePlugin'
+
+export type LexicalContentEditableProps = {
+  editorRef: RefObject<LexicalEditor>
+  contentEditableRef: RefObject<HTMLDivElement>
+  onChange?: (content: SerializedEditorState) => void
+  onSubmit?: () => void
+  onFocus?: () => void
+  onMentionNodeMutation?: (mutations: NodeMutations<MentionNode>) => void
+  initialEditorState?: InitialEditorStateType
+  autoFocus?: boolean
+  plugins?: {
+    vaultChat?: {
+      onVaultChat: () => void
+    }
+    templatePopover?: {
+      anchorElement: HTMLElement | null
+    }
+  }
+}
+
+export default function LexicalContentEditable({
+  editorRef,
+  contentEditableRef,
+  onChange,
+  onSubmit,
+  onFocus,
+  onMentionNodeMutation,
+  initialEditorState,
+  autoFocus = false,
+  plugins,
+}: LexicalContentEditableProps) {
+  const app = useApp()
+
+  // // initialize editor state
+  // useEffect(() => {
+  //   if (initialContent) {
+  //     updaterRef.current?.update(initialContent)
+  //   }
+  //   // eslint-disable-next-line react-hooks/exhaustive-deps
+  // }, [])
+
+  const initialConfig: InitialConfigType = {
+    namespace: 'ChatUserInput',
+    theme: {
+      root: 'smtcmp-chat-input-root', // FIXME: change class name
```

Comment:
Why was this comment added?

Reply:
Changed class name from `smtcmp-chat-input-...` to `smtcmp-lexical-content-editable-...`